pkg/cloud/gcp/actuators/machine/actuator.go

```diff
@@ -47,10 +68,17 @@ func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machi
 		machine:       machine,
 	})
 	if err != nil {
-		return fmt.Errorf(scopeFailFmt, machine.Name, err)
+		fmtErr := fmt.Errorf(scopeFailFmt, machine.Name, err)
+		return a.handleMachineError(machine, apierrors.InvalidMachineConfiguration("error decoding MachineProviderConfig: %v", fmtErr), createEventAction)
```

Comment:
why is this passing using `InvalidMachineConfiguration`
and `error decoding MachineProviderConfig`?? 
Is there any value on handleMachineError? wouldn't it be simpler just `eventf` here - Same question for all the functions

Reply:
Since `Actuator.handleMachineError` logs error and records the same error as event at the same time.